pinot-common/src/main/java/org/apache/pinot/common/data/FieldSpec.java

```diff
@@ -98,15 +102,21 @@ public FieldSpec(String name, DataType dataType, boolean isSingleValueField) {
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, @Nullable Object defaultNullValue) {
     this(name, dataType, isSingleValueField, DEFAULT_MAX_LENGTH, defaultNullValue);
   }
-
+  
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
       @Nullable Object defaultNullValue) {
+    this(name, dataType, isSingleValueField, maxLength, defaultNullValue, null);
+  }
+  
+  public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
+      @Nullable Object defaultNullValue, @Nullable String objectType) {
```

Comment:
Shouldn't this method actually set _objectType?

Reply:
yes, will fix it.